### PR TITLE
ci: fix diverging PRs

### PR DESCRIPTION
**Pull request by oldgiova**
In some conditions the upstream pr branch could diverge from the local branch. With this fix it should rebase the upstream changes locally.
For example: https://gitlab.com/Northern.tech/Mender/mender-helm/-/jobs/8049202233#L251

### CODE REVIEW

**Comment by oldgiova** [comment]
A possible alternative is the original behavior: just clone from scratch the github repo into a temp folder every time, and add modification into this, instead using the preexisting local `.git`.

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -863,6 +863,7 @@ changelog:
     - test -z "$RELEASE_PLEASE_PR" && echo "No release-please PR found" && exit 0
     - echo "INFO - generating changelog notes from git cliff"
         && gh pr checkout $RELEASE_PLEASE_PR
+        && git pull
```
Comment (alfrunes) [inline .gitlab-ci.yml:866]: Couldn't this potentially cause merge conflicts?
What about a simple `git fetch github-${TS_SUFFIX} pull/$RELEASE_PLEASE_PR/head && git reset --hard FETCH_HEAD`?

Comment (oldgiova) [reply .gitlab-ci.yml]: yeah, thanks for the trick!

**Review by alfrunes — APPROVED** [review]
Just a side comment to #348; maybe it would be cleaner to simply ignore the error when adding the remote instead of carrying around the `TS_SUFFIX` env variable.
```bash
git remote add github https://${GITHUB_USER}:${GITHUB_BOT_TOKEN_REPO_FULL}@github.com/${GITHUB_REPO_URL} || true
```

**Pull request merged**